src/features/auth/components/Signup.tsx

```diff
@@ -0,0 +1,189 @@
+import {Form} from "@/components/shadcn-ui/form";
+import {FC} from "react";
+import {SubmitHandler, useForm} from "react-hook-form";
+import {RegisterUser, registerUser} from "../schema";
+import {zodResolver} from "@hookform/resolvers/zod";
+import {z} from "zod";
+
+export const Signup: FC = () => {
+  const {register, handleSubmit} = useForm<RegisterUser>({
+    defaultValues: {
+      firstname: "",
```

Comment:
What are these default values for ?
There is already a validator schema, and some prop should be null to avoid updating them